### PR TITLE
Add support for Mac OS

### PR DESCRIPTION
**Pull request by zhijian-liu**
This PR adds support for Mac OS. The main change is to replace `long long` with `int64_t` and `unsigned long long` with `uint64_t`.

### CODE REVIEW

File: torchsparse/backend/hash/hash_cpu.cpp [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 #include <vector>
 
-void cpu_hash_wrapper(int N, const int *data, long *out) {
+void cpu_hash_wrapper(int N, const int *data, int64_t *out) {
```
Comment (CCInc) [inline torchsparse/backend/hash/hash_cpu.cpp:7]: I think `long` fits into `int32_t` but it should make no difference.

**Review by CCInc — APPROVED** [review]
Looks good! The only thing I would check is if you want to change the `long` values to `int32_t`. I think `long` is default 32 bits, and `long long` is 64.

**Comment by zhijian-liu** [comment]
That's a good point. Not sure why we were using `long int` rather than `long long` before (@kentangSJTU). From my understanding, `out` should be a 64-bit tensor.

**Comment by zhijian-liu** [comment]
I have switched these `long` to `int64_t`, which is also aligned with Pytorch's definition (https://github.com/pytorch/pytorch/blob/master/c10/core/ScalarType.h#L33).

**Review by kentangSJTU — APPROVED** [review]

**Pull request merged**